modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4IncrementalRequestHandlingIT.java

```diff
@@ -501,6 +514,72 @@ private void assertHttpBodyLogging(Consumer<ClientContext> test) throws Exceptio
         }
     }
 
+    public void testBulkIndexingRequestSplitting() throws Exception {
+        final var watermarkBytes = between(100, 200);
+        final var tinyNode = internalCluster().startCoordinatingOnlyNode(
+            Settings.builder()
+                .put(IndexingPressure.SPLIT_BULK_LOW_WATERMARK.getKey(), ByteSizeValue.ofBytes(watermarkBytes))
+                .put(IndexingPressure.SPLIT_BULK_LOW_WATERMARK_SIZE.getKey(), ByteSizeValue.ofBytes(watermarkBytes))
+                .build()
+        );
+
+        try (var clientContext = newClientContext(tinyNode, cause -> ExceptionsHelper.maybeDieOnAnotherThread(new AssertionError(cause)))) {
+            final var request = new DefaultHttpRequest(HTTP_1_1, POST, "/_bulk");
+            request.headers().add(CONTENT_TYPE, APPLICATION_JSON);
+            HttpUtil.setTransferEncodingChunked(request, true);
+
+            final var channel = clientContext.channel();
+            channel.writeAndFlush(request);
+
+            final var indexName = randomIdentifier();
+            final var indexCreatedListener = ClusterServiceUtils.addTemporaryStateListener(
+                cs -> Iterators.filter(
+                    cs.metadata().indicesAllProjects().iterator(),
+                    indexMetadata -> indexMetadata.getIndex().getName().equals(indexName)
+                ).hasNext()
+            );
+
+            indexCreatedListener.addListener(ActionListener.running(() -> logger.info("--> index created")));
+
+            final var valueLength = between(10, 30);
+            final var docSizeBytes = "{'field':''}".length() + valueLength;
+            final var itemCount = between(watermarkBytes / docSizeBytes + 1, 300); // enough to split at least once
```

Comment:
Would we not risk the last item(s) go into `lastItems` and thus there is not splitting (just challenging the "at least once" part of the comment).

I think there is a minimum size too for netty to pass along data, like 8kb IIRC? So if the bulk is smaller than that we still see only one chunk?

Reply:
There'll always be a split: we won't go down the `lastItems` path until after sending `DefaultLastHttpContent`, by which point we've already definitely created the index.

8kiB is the maximum chunk size but if Netty has less than that available then it still sends what it has. It doesn't delay until it has a whole 8kiB chunk.